utils/truncate/query_test.go

```diff
@@ -38,8 +38,8 @@ func TestQuery(t *testing.T) {
 		"абвгдеё": {"а ...", true},
 
 		// Unicode replacement characters
```

Comment:
please add comment here, why \uFFFD doesn't work for us.

Reply:
Yeah, I wonder why it doesn't work for you.

`[]rune(string)` converts a binary string to Unicode runes, replacing each sequence of bytes that can't be a valid Unicode rune with the [Unicode replacement character](https://en.wikipedia.org/wiki/Specials_(Unicode_block)#Replacement_character) (which itself is a valid Unicode rune). That what this function did, that's what the comment below means, and that's why this behavior was tested. Those are not "invalid chars".

So I wonder why you decided to strip them completely instead of replacing them by a rune specifically designed for that case.

Reply:
When it was replaced by \uFFFD it still caused issue with invalid UTF-8 char, even it is valid UTF-8 char. Maybe it is problem of conversion in diffrent location.

Reply:
Please make a test that shows that encoding U+FFFD in protobuf string produces an error

Reply:
AFAIK, even with these characters we are still having the same error.

Reply:
I am not sure if I can do it in tests. To test it I always release binary and then test it this way. Here are steps which are needed to reproduce it. 

Maybe, this is similar problem even with valid UTF-8 chars:
https://bugs.mysql.com/bug.php?id=78758

Reply:
> Here are steps which are needed to reproduce it.

Where?

MySQL bug doesn't involve protobuf, so I'm not sure how that can be related.

I'm sure you can write a test that shows that protobuf can't encode U+FFFD – just encode that string with protobuf and check for error. That's less than 10 lines.

Reply:
@AlekSi Test passed, also I tried again binary without ToValidUTF8 (only with truncate) method and it is fine. For first time I used probably wrong binary. So you was right and it is not necessary. I will remove it and modify tests for it.

Reply:
> Test passed

Please add it there